### DISCUSSION TITLE
Pull up request for a change in Core and DSP

### DISCUSSION THREAD

**Discussion by gildadam** [ARM-software/CMSIS_5#1663] [General]
Hi there, I am looking to create a change that affects a header included in the Core folder as well as a header that was originally from the DSP folder. These files are:
Core/Include/cmsis_compiler.h
DSP/Include/arm_math_types.h

However, I see that the develop branch does not have both of these folders, and instead has a separate repo for DSP. My end goal is to see this change reflected in master, which is where the company I work for pulls its external repo from. How could I go about doing this? Would I be able to create a direct pull request for master, or must I create one for the DSP repo, and one for the CMSIS-5 repo, and then somehow both of those changes will be merged to master?

**Comment by jkrech** [comment]
We have moved on to CMSIS Version 6.
At this point there are no plans for updating CMSIS Version 5.
As you can see here: https://arm-software.github.io/CMSIS_6/latest/General/index.html
we moved away from a big pack towards multiple smaller sized packs and github repositories.

cmsis_compiler.h - https://github.com/ARM-software/CMSIS_6/tree/main/CMSIS/Core/Include
arm_math_types.h - https://github.com/ARM-software/CMSIS-DSP/tree/main/Include

There is nothing that needs merging.

**Comment by gildadam** [comment]
Our company's external branch tracks the master branch of CMSIS-5 so we would need the changes in there, what would you recommend we do?

**Comment by jkrech** [comment]
I am just helping out while Jonatan is out of office: @JonatanAntoni could you please provide some feedback/recommendations. Thanks.

**Comment by gildadam** [comment]
Okay thank you, will wait on Jonatan's response.

**Comment by gildadam** [comment]
Hi there,
when will Jonatan be back in office and able to assist? Fyi, our company branch that pulls from CMSIS-5 is Garmin Ltd's external branch.

**Comment by JonatanAntoni** [comment]
Hi @gildadam,

May I ask you to check if moving to CMSIS 6 and separate CMSIS-DSP is an option for you?
You'd need to add multiple external's then, depending on the components you are relying on. E.g., RTX5 has been offloaded into its own repository as well.

-Jonatan

**Comment by gildadam** [comment]
Hi @JonatanAntoni,

After some discussion, we will be creating CMSIS-6 and CMSIS-DSP externals. Of course, this will take some time to navigate the different approvals, but since CMSIS-5 is no longer supported, we will just push the changes needed to our external repo in the meantime. Thank you for the assistance!

Adam
